src/fields.jl

```diff
@@ -86,11 +86,14 @@ Interpolates a 1D `input` field that lives on `Face`s to `output` field that liv
 """
 function 𝐼x!(output::Field1D{Centre}, input::Field1D{Face, Grid1D{Periodic}})
     nx = input.grid.nx
-        
-    output.data[nx] = (input.data[1] + input.data[nx]) / 2
-    for i in 1:nx-1
+    
+    for i in 1:nx
         output.data[i] = 𝐼xᶜ(i, input)
     end
+    
+    fill_halos!(output)
+    
+    return nothing
 end
```

Comment:
What if the user defines hx = 0? Then, we need this:
`output.data[nx] = (input.data[1] + input.data[nx]) / 2`.

Reply:
Perhaps `hx=0` shouldn't be allowed. For periodic boundary conditions you need *at least* 1 halo point.